### ISSUE TITLE
snapscraft builds

### ISSUE THREAD

**Issue by joeblew99** [mh-cbon/go-github-release#1]
this is an awesome project.

Could you have a look at implementing snap based builds. These are for ubuntu but other linux distros are also either suppoting them, or starting to support them.

This explains it all.
https://github.com/syncthing/syncthing/pull/3636

In here you can see their project go build script that calls to make a snapscraft build.
https://github.com/syncthing/syncthing/blob/master/build.go

Really curiosu what you think..

**Comment by mh-cbon** [comment]
totally agree with you.

Note that snap is not the only new packer for linux. Gnome is pushing its on new one too, [flatpack](http://flatpak.org/).

At the time i worked on that project i could not find enough resources to jump into it.

**Comment by mh-cbon** [comment]
something that may be useful is to try to start a common group about those questions. I think they are yet several solutions to provide an installer to various os, but they are not coordinated.

**Comment by joeblew99** [comment]
i agree this is the problem. I guess i was seeing what your up to and thinking "hey here is all the bits to do it in this git org :) ".

But heres an approach:
I think the best place for this is over at Gallium. Because its a pure golang based alternative to Electron - no nodejs needed. For all application bundling, updating, etc we make a seperate repo called "gallium-pack". Gallium can use it but also anyone else. I can ask the org maintainer.

**Comment by mh-cbon** [comment]
I can only tell that you seem very enthusiast about gallium (i agree about that), 

I d want just to point that go has the ability to build 
- qt app (and some other various bindings to gui toolkits)
- mobile apps
- cli app
- web app

IMVHU, such gallium-pack command, ideally, is a wrapper which applies gallium conventions&configuration to a series of per-target tool which will realize the desired packing.

Maybe gallium is the definitive solution that will simplify the problem of desktop applications at once, in which case doing a specific worth to be done, maybe its just a solution among many other, in which case a more agnostic solution may be more valuable to everyone. Where are you heading to ?

**Comment by joeblew99** [comment]
Hey again,

Yes i agree with you about go being able to build QT, mobile, CLI and web. Its exactly what i am working on making much more standardised. Gallium is just what i am using fro Desktop.

Regarding your idea for a place to put all this together, THIS repo is a very nice place to document it as you have done already with a link to all the toosl that do a specific thing. 

I have started working on an Update system for Apps deployed to Servers and Desktops. 
I have also started working on a Cloud based development environment.

I just send you an email about it :)

**Comment by mh-cbon** [comment]
ok ill check this out asap.

**Comment by mh-cbon** [comment]
https://snapcraft.io/create/

https://github.com/snapcore/snapd

**Comment by mh-cbon** [comment]
https://github.com/goreleaser/goreleaser#snapcraft-build-customization
